lib/email/receiver.rb

```diff
@@ -156,6 +164,11 @@ def process_internal
         raise UserNotFoundError unless SiteSetting.enable_staged_users
       end
 
+      recipients = get_all_recipients(@mail)
```

Comment:
I feel kinda bad we're doing all this work (decoding all the recipients) to just get a count and then throwing it away. And then later, we're re-doing this work to add "staged" users.

Any chances we could keep those recipients and re-use them later?

Reply:
The operation that iterates over the email addresses in the email is not expensive. I wanted to do what you suggested, but the class is complex and the other method that iterates over the email addresses is used for with another email object. I can still do it, but the effort of storing the list of emails is larger than just iterating over them.

Reply:
Right, `add_other_addresses` is used for the current email and the embedded email in the case we've received a fwed email.